assisted-inject-processor/src/main/java/com/squareup/inject/assisted/processor/AssistedInjectProcessor.kt

```diff
@@ -244,7 +252,7 @@ class AssistedInjectProcessor : AbstractProcessor() {
       throw StopProcessingException("Factory interface defines multiple factory methods.", factory)
     }
     val method = methods.single()
-    if (!types.isAssignable(type.asType(), method.returnType)) {
+    if (!types.isAssignable(types.erasure(type.asType()), types.erasure(method.returnType))) {
```

Comment:
```java
types.isSameType(type.asType(), method.returnType)
```
returns `false` too. I don't see anything in auto-common either...

Maybe @ronshapiro knows?